idl/activity_forecast_evol.pro

```diff
@@ -0,0 +1,114 @@
+;+
+;
+; Name        : activity_forecast_evol
+;
+; Purpose     : Calculates flaring probabilities from flaring rates
+;               using the 24-hr evolution of McIntosh Classifications
+;
+; Syntax      : activity_forecast_evol, summary, names, cprob, mprob, xprob
+;
+; Inputs      : summary = output from ar_comb.pro
+; 
+; Outputs     : cprob, mprob, xprob: array of probabilities for each
+;               active region
+;
+; Examples    : IDL> activity_forecast, output_path, summary, names,
+;                    mci, cprob, mprob, xprob
+;
+; History     : March 2017 - Written by Aoife McCloskey
+;
+; Contact     : mccloska@tcd.ie (Aoife McCloskey, TCD)
+;
+;-
+
+
+pro activity_forecast_evol,output_path,summary, names, mci, cprob_evol, mprob_evol, xprob_evol
+
+
+; Defining McIntosh Classifications
+  Zur = ['A','B','H', 'C', 'D', 'E', 'F']
+  Pen= ['X', 'R', 'S', 'A','H', 'K']
+  Comp=['X', 'O', 'I' , 'C']
+
+; Restoring sav file for McIntosh Evolution flaring rates (uses NOAA data from 1988-2008)
+  flarehist_file = getenv('WORKING_PATH')+'/data/mcint_evol_flarehist.sav'
+  restore, flarehist_file
+
+; Read McIntosh classifications for current regions today and yesterday
+
+  mci   = strupcase( reform( summary[ 3, * ] ))
```

Comment:
wouldn't be better to get a function that converts `summary` into a structure, so we don't need to parse each time we want to read the summary?